calcom/cal.com
# fix: Fix 'Book a new time' link in request-reschedule for Team Event

## What does this PR do?

Fixes #12147 #12141

Also, adds tests for `requestReschedule` handler. Restructures bookingScenario test dependencies as per the requirements

## Type of change

## How should this be tested?
- Create a team event
- Book that
- Login as organizer and do a request-reschedule
- See the email that you received with 'Book a new time' link\
- Clicking on it should take you to the correct reschedule page. Earlier it was taking the user to team profile page and he was unable to reschedule.



## Mandatory Tasks


## Checklist
- I haven't added tests that prove my fix is effective or that my feature works
